AutoGear.lua

```diff
@@ -3935,10 +3939,13 @@ function AutoGearGetBest1hPairing(info)
 end
 
 function AutoGearTooltipHook(tooltip)
+	-- Added nil check to avoid errors which were happening on GetItem() call
+	if tooltip ~= GameTooltip or tooltip == nil then return end
```

Comment:
We previously hooked 4 tooltips, so this should either check for all the previous tooltip types or check the new Enum for the tooltip data type `Item`.  There may be a way to check that.

Reply:
As per the API changes, it seems the previous 4 hooks are handled by the new single line. The first parameter of the new handler determines what type of tooltips to hook to and works globally. It was my understanding the one line now does the same as the old 4 hooks. I could be mistaken and I am also guessing here as to the intention of the addon as a whole.

Tooltip Script Handlers

The following GameTooltip script handlers have been removed. This change will not be reflected in the UI XML schema definition until a future patch.

[OnTooltipAddMoney](https://wowpedia.fandom.com/wiki/UIHANDLER_OnTooltipAddMoney)
OnTooltipSetAchievement
OnTooltipSetEquipmentSet
[OnTooltipSetItem](https://wowpedia.fandom.com/wiki/UIHANDLER_OnTooltipSetItem)
OnTooltipSetQuest
OnTooltipSetSpell
OnTooltipSetUnit
Usages of these script handlers should be replaced by registering callbacks with the new [TooltipDataProcessor.AddTooltipPostCall](https://github.com/Ketho/wow-ui-source-df/search?q=AddTooltipPostCall) function. Note that callbacks registered with this mechanism are global and will be triggered for all tooltips which inherit from GameTooltipTemplate.

local function OnTooltipSetItem(tooltip, data)
    if tooltip == GameTooltip then
        print("OnTooltipSetItem", tooltip, data)
    end
end

-- Replace 'Enum.TooltipDataType.Item' with an appropriate type for the tooltip
-- data you are wanting to process; eg. use 'Enum.TooltipDataType.Spell' for
-- replacing usage of OnTooltipSetSpell.
--
-- If you wish to respond to all tooltip data updates, you can instead replace
-- the enum with 'TooltipDataProcessor.AllTypes' (or the string "ALL").

TooltipDataProcessor.AddTooltipPostCall(Enum.TooltipDataType.Item, OnTooltipSetItem)

Reply:
What I mean is that we want AutoGear to operate on the ShoppingTooltip1, ShoppingTooltip2, and ItemRefTooltip, if those still exist at all.  The logic in the line upon which this review comment thread is based excludes those tooltips specifically by checking for `GameTooltip` only, instead of checking `Enum.TooltipDataType.Item`, if that is possible to check.

Reply:
I did some digging and found lua and xml code from the 10.0.2 beta which shows GameTooltip and ItemRefTooltip inherits from GameTooltipTemplate so the new hook should cover that. ShoppingTooltip* inherit from ShoppingTooltipTemplate which I can't find anything on so I plan to try to return those previous hooks to see what that does.

https://github.com/tomrus88/BlizzardInterfaceCode/blob/168d22d05fdaf7d3ca2e1af0d02a5cd9707b15b5/Interface/FrameXML/GameTooltip.xml

Reply:
I think I finally understand what you were getting at. I tried to add the ShoppingTooltip* hooks and that threw an error, that was when I realized the issue wasn't the new hook but my if check. I updated that for my code and will have that with my next checkin.

if tooltip ~= GameTooltip or tooltip ~= ShoppingTooltip1 or tooltip ~= ShoppingTooltip2 or tooltip ~= ItemRefTooltip or tooltip == nil then return end